src/nus/cs2103/addressbook/AddressBook.java

```diff
@@ -557,53 +641,60 @@ private static void updateLatestViewedPersonListing(List<String[]> newListing) {
      */
 
     /**
-     * Create storage file if it does not exist. Shows feedback to user.
+     * Creates storage file if it does not exist. Shows feedback to user.
      *
      * @param filePath file to create if not present
      */
-    private static void tryToCreateFileIfMissing(String filePath) {
+    private static void createFileIfMissing(String filePath) {
         final File storageFile = new File(filePath);
+        if (storageFile.exists()) {
+            return;
+        }
+
+        showToUser(String.format(MESSAGE_ERROR_MISSING_STORAGE_FILE, filePath));
+
         try {
-            if (storageFile.createNewFile()) {
-                showToUser(String.format(MESSAGE_ERROR_MISSING_STORAGE_FILE, filePath));
-                showToUser(String.format(MESSAGE_STORAGE_FILE_CREATED, filePath));
-            }
+            storageFile.createNewFile();
+            showToUser(String.format(MESSAGE_STORAGE_FILE_CREATED, filePath));
         } catch (IOException ioe) {
             showToUser(String.format(MESSAGE_ERROR_CREATING_STORAGE_FILE, filePath));
+            exitProgram();
         }
     }
 
     /**
-     * Converts contents of a file into a list of persons. Shows feedback to user.
+     * Converts contents of a file into a list of persons.
+     * Exits program if any errors in reading or decoding was encountered.
      *
      * @param filePath file to load from
-     * @return able to decode: optional containing list of persons in file
-     *         problems decoding: empty optional
+     * @return the list of decoded persons
      */
-    private static Optional<List<String[]>> loadPersonsFromFile(String filePath) {
-        try (final BufferedReader storageReader =
-                     new BufferedReader(new FileReader(filePath))) {
-            final List<String[]> decodedPersons = new ArrayList<>();
+    private static List<String[]> loadPersonsFromFile(String filePath) {
+        final List<String[]> decodedPersons = new ArrayList<>();
+        //TODO: separate file reading from decoding
```

Comment:
It works, but I cannot extract the {decode list of strings into list of persons} as a separate method because there is no nice way to pass exception information back for user feedback.